tests/epyccel/test_epyccel_variable_annotations.py

```diff
@@ -221,7 +219,7 @@ def homogeneous_set_annotation ():
 
 def test_homogeneous_empty_list_annotation_int(language):
     def homogeneous_list_annotation():
-        a: 'list[int]'  # pylint: disable=unused-variable
+        a: 'list[int]'
```

Comment:
Can we not stop using the string?

Reply:
Yes we can!

Reply:
I will take some out of strings but not all to ensure we keep testing strings. We cannot get rid of string annotations completely as there is no satisfactory replacement for NumPy arrays. Strings are also needed for class methods e.g.:
```python
class A:
    def __add__(self, other : 'A'):
        ...
```

Reply:
Fixed 044f81a88

Reply:
Yes, we should test strings for the cases that are not covered otherwise!

Should we keep allowing for string annotations in the cases where we have a clean alternative like sets, lists, dicts and tuples?

Reply:
I think we have to because otherwise you have a problem if you want e.g. `tuple[float[:], float[:,:]]`. It makes things easier for the stub files too as I don't have to work out whether or not the final type needs to be in quotes or not.

It doesn't add too much to the code as once the string annotation is parsed the string is discarded and we handle `list[int]` and `'list[int]'` identically.